src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -135,7 +142,10 @@ public void visitJavadocToken(DetailNode ast) {
         String firstSentence = getFirstSentence(ast);
         final int endOfSentence = firstSentence.lastIndexOf(period);
         if (endOfSentence == -1) {
-            if (!firstSentence.trim().startsWith("{@inheritDoc}")) {
+            if (!firstSentence.trim().startsWith(INHERIT_DOC_LITERAL)
+                && !FIRST_CHARACTER_ASTERIC_PATTERN.matcher(firstSentence.trim())
+                .replaceFirst("").trim().startsWith(INHERIT_DOC_LITERAL)
```

Comment:
indent this line further to right.

This IF expression is hard to read.... please think how to make it better. Code should be read as text.